src/gtfs/read.rs

```diff
@@ -143,6 +145,113 @@ impl From<Stop> for objects::StopPoint {
     }
 }
 
+#[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq, Hash)]
+enum RouteType {
+    #[allow(non_camel_case_types)]
+    Tramway_LightRail,
+    Metro,
+    Rail,
+    Bus,
+    Ferry,
+    CableCar,
+    #[allow(non_camel_case_types)]
+    Gondola_SuspendedCableCar,
+    Funicular,
+    Other(u16),
+}
+
+impl RouteType {
+    fn to_gtfs_value(&self) -> String {
+        match self {
+            &RouteType::Tramway_LightRail => "0".to_string(),
+            &RouteType::Metro => "1".to_string(),
+            &RouteType::Rail => "2".to_string(),
+            &RouteType::Bus => "3".to_string(),
+            &RouteType::Ferry => "4".to_string(),
+            &RouteType::CableCar => "5".to_string(),
+            &RouteType::Gondola_SuspendedCableCar => "6".to_string(),
+            &RouteType::Funicular => "7".to_string(),
+            &RouteType::Other(i) => i.to_string(),
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+struct Route {
+    #[serde(rename = "route_id")]
+    id: String,
+    agency_id: Option<String>,
+    #[serde(rename = "route_short_name")]
+    short_name: String,
+    #[serde(rename = "route_long_name")]
+    long_name: String,
+    #[serde(rename = "route_desc")]
+    desc: Option<String>,
+    #[serde(deserialize_with = "de_route_type")]
+    route_type: RouteType,
+    #[serde(rename = "route_url")]
+    url: Option<String>,
+    #[serde(rename = "route_color", default, deserialize_with = "string_to_color")]
+    color: Option<objects::Rgb>,
+    #[serde(rename = "route_text_color", default, deserialize_with = "string_to_color")]
+    text_color: Option<objects::Rgb>,
+    #[serde(rename = "route_sort_order")]
+    sort_order: Option<u32>,
+}
+
+impl Route {
+    fn is_same_line(&self, other: &Route) -> bool {
+        if self.agency_id == other.agency_id {
+            if self.short_name != "" {
+                if self.short_name == other.short_name {
+                    return true;
+                }
+            } else if self.long_name == other.long_name {
+                return true;
+            }
+        }
+        return false;
+    }
+}
+
+fn de_route_type<'de, D>(deserializer: D) -> Result<RouteType, D::Error>
+where
+    D: ::serde::Deserializer<'de>,
+{
+    use serde::Deserialize;
+    let mut i = u16::deserialize(deserializer)?;
+    if i > 7 && i < 99 {
+        i = 3;
```

Comment:
Shouldn't you error in this case?

Reply:
I didn't want to just panic with an invalid gtfs, but make a fallback on bus mode.

Reply:
At least `err!("illegal route_type: {}, using 3 as fallback")`